source/common/tcp_proxy/tcp_proxy.cc

```diff
@@ -362,34 +362,28 @@ Network::FilterStatus Filter::initializeUpstreamConnection() {
     return Network::FilterStatus::StopIteration;
   }
 
-  if (auto downstream_connection = downstreamConnection(); downstream_connection != nullptr) {
```

Comment:
The removal of this nullptr check worried me, but if all tests pass and there are no comments explaining why this nullptr check is needed I have to guess that this change is fine.